src/core/lib/uri/uri_parser.h

```diff
@@ -40,15 +38,21 @@ class URI {
     bool operator==(const QueryParam& other) const {
       return key == other.key && value == other.value;
     }
+    bool operator<(const QueryParam& other) const {
```

Comment:
Why is this added? Arbitrary query parameters can't be reordered. If you just sorted by the key and left values in their existing order, that'd probably be mostly fine, but sorting within a key is definitely asking to break something. But I don't actually see any usage of this. (So I can't check if duplicates are wrongly removed.)

Reply:
Seems like it is prone to abuse/misunderstanding here and it'd be trivial to define in the xds code instead. So I don't really understand, but I don't feel the need to discuss it more.